### PR TITLE
Edits gridscore stats

### PR DESCRIPTION
**Pull request by horsto**
Worked a bit on the gridscore stats function. 
- Fixed some imports
- Changed output a bit 
- More checkpoints
- Changed to SEP for field extraction since `local_maxima` and `max_tree_local_maxima` were too inconsistent

### CODE REVIEW

File: opexebo/analysis/gridscore.py [hunk 1]
```diff
@@ -78,58 +79,56 @@ def gridscore(aCorr, **kwargs):
...
-    
     # normalize aCorr in order to find contours
     aCorr = aCorr / aCorr.max()
-    cFieldRadius = int( np.floor(_findCentreRadius(aCorr, **kwargs)) )
+    centre = -0.5 + np.array(aCorr.shape)/2 # centre : also [y, x]
+    cFieldRadius = int( np.floor(_findCentreRadius(aCorr)) )
     if debug:
         print("Center radius is {}".format(cFieldRadius))
 
-    if cFieldRadius in [-1, 0, 1]:# or cFieldRadius[0] > .8*aCorr.shape[0]/2 or cFieldRadius[0] > .8*aCorr.shape[1]/2:
```
Comment (simon-ball) [inline opexebo/analysis/gridscore.py]: This was based on a previous addition of yours - no longer necessary?

File: opexebo/analysis/gridscore.py [hunk 2]
```diff
@@ -148,124 +147,191 @@ def gridscore(aCorr, **kwargs):
...
-        
-        # Bearing from field to centre
+    gs_ellipse_theta, gs_ellipse, gs_aspect_ratio = np.nan, np.nan, np.nan
+    gs_orientations_std, gs_orientation = np.nan, np.nan
+    gs_positions = np.array([np.nan, np.nan, np.nan, np.nan, np.nan, np.nan])
+    gs_orientations = np.array([np.nan, np.nan, np.nan])
+    gs_spacings = np.array([np.nan, np.nan, np.nan])
+
+    # Find fields in autocorrelogram
+    acorr_fields = sep.extract(aCorr, mask=mask, thresh=.1) # CONSIDER MAKING THRESHOLD AN ARGUMENT
```
Comment (simon-ball) [inline opexebo/analysis/gridscore.py]: I need to test this against the ephys data before making it the mandatory method - previously, it couldn't cope, and I still haven't figured out why

File: opexebo/analysis/gridscore.py (hunk 1 above)
Comment (horsto) [reply opexebo/analysis/gridscore.py]: This did not seem to be necessary anymore (based on > 6000 example cells from imaging data).

File: opexebo/analysis/gridscore.py (hunk 2 above)
Comment (horsto) [reply opexebo/analysis/gridscore.py]: Thanks, yes, I would be grateful if you could compare with ephys data. Would be great if we could use the same method here.

File: opexebo/analysis/gridscore.py [hunk 3]
```diff
@@ -148,124 +147,191 @@ def gridscore(aCorr, **kwargs):
...
+        positions, distance, orientation = all_coords[sorted_ids_dist], distance[sorted_ids_dist], orientation[sorted_ids_dist]
+        # ... then re-sort remaining fields by angle
+        sorted_ids_ang = np.argsort(orientation)
+
+        ################# GATHER OUTPUT #################
+        gs_positions = positions[sorted_ids_ang]
+        # For grid orientation and spacing take only 3 out of 6 neighbouring fields
+        gs_spacings = distance[sorted_ids_ang][:3]
+        gs_orientations = np.degrees(orientation[sorted_ids_ang][:3]) % 180
+
```
Comment (simon-ball) [inline opexebo/analysis/gridscore.py]: How do you exclude the central field at dist=0?

Comment (horsto) [reply opexebo/analysis/gridscore.py]: I am masking it before I detect fields (mask is created in line 153 of gridscore.py for both > best grid score radius and < center field radius).

**Pull request merged**